tests/run-build.sh

```diff
@@ -10,53 +10,81 @@ if test -e ci-skip ; then
     exit 0
 fi
 
-# Setup go environment on semaphore
-if [ -f /opt/change-go-version.sh ]; then
-    . /opt/change-go-version.sh
-    change-go-version 1.4
-fi
-
+SRC_CHANGES=1 # run functional tests by default
+DOC_CHANGES=1 # process docs by default
 RKT_STAGE1_USR_FROM="${1}"
 RKT_STAGE1_SYSTEMD_VER="${2}"
-
 BUILD_DIR="build-rkt-${RKT_STAGE1_USR_FROM}-${RKT_STAGE1_SYSTEMD_VER}"
 
+if [ "${CI-}" == true ] ; then
+	# https://semaphoreci.com/docs/available-environment-variables.html
+	if [ "${SEMAPHORE-}" == true ] ; then
+        # We might not need to run functional tests
+        SRC_CHANGES=$(git diff-tree --no-commit-id --name-only -r HEAD..origin/master | \
+            grep -cEv "^Documentation/")
+        # We might not need to process the docs
+        DOC_CHANGES=$(git diff-tree --no-commit-id --name-only -r HEAD..origin/master | \
+            grep -cE "^Documentation/")
+
+        # Setup go environment on semaphore
+        if [ -f /opt/change-go-version.sh ]; then
+            . /opt/change-go-version.sh
+            change-go-version 1.4
+        fi
+    fi
+fi
+
+if [ ${SRC_CHANGES} -eq 0 ] && [ ${DOC_CHANGES} -eq 0 ]; then
+    echo No changes detected.
+    exit 0
+fi
+
+# In case it wasn't cleaned up
+if [ -e "${BUILD_DIR}" ]; then
```

Comment:
I'd say that this `if` is unnecessary. Maybe just call `sudo rm -rf "${BUILD_DIR}"` directly.